html/modules/custom/rwr_sitrep/src/Controller/PdfController.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Drupal\rwr_sitrep\Controller;
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\State\StateInterface;
+use Drupal\group\Entity\GroupRelationship;
+use Drupal\node\Entity\Node;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Page controller for tabs.
+ *
+ * Most of this taken from https://github.com/UN-OCHA/dsr-site/blob/develop/
+ * html/modules/custom/dsr_custom/src/Controller/PdfController.php.
+ */
+class PdfController extends ControllerBase {
+
+  /**
+   * The state store.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * Language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('state'),
+      $container->get('languageManager')
+    );
+  }
+
+  /**
+   * PdfController constructor.
+   *
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The state controller.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $languageManager
+   *   The language manager.
+   */
+  final public function __construct(StateInterface $state, LanguageManagerInterface $languageManager) {
+    $this->state = $state;
+    $this->languageManager = $languageManager;
+  }
+
+  /**
+   * Access check for sitreps.
+   */
+  public function isSitrep(Node $node) : AccessResult {
```

Comment:
Get the group from the relation, check if `field_group_type` exists and check `field_pdf_enabled` on that term.

This will make the code more generic.

@todo: Add `field_pdf_enabled` to the terms